ooniprobe/Model/NetworkMeasurement.mm

```diff
@@ -509,24 +540,28 @@ -(void)on_entry:(const char*)str{
         NSData *data = [[NSString stringWithUTF8String:str] dataUsingEncoding:NSUTF8StringEncoding];
         NSDictionary *json = [NSJSONSerialization JSONObjectWithData:data options:0 error:&error];
         int blocking = ANOMALY_GREEN;
-        
+        if (error != nil) {
+#ifdef DEBUG
+            NSLog(@"Error parsing JSON: %@", error);
+#endif
+            blocking = ANOMALY_ORANGE;
+            [self updateAnomaly:blocking];
+            return;
+        }
         //whatsapp_endpoints_status and whatsapp_web_status and registration_server_status must be both false if null (set test anomaly), if at least one is true (set test blocked)
         NSArray *keys = [[NSArray alloc] initWithObjects:@"whatsapp_endpoints_status", @"whatsapp_web_status", @"registration_server_status", nil];
         for (NSString *key in keys) {
-            if ([json objectForKey:key]){
-                if ([json objectForKey:key] == [NSNull null]) {
+            if ([[json objectForKey:@"test_keys"] objectForKey:key]){
+                if ([[json objectForKey:@"test_keys"] objectForKey:key] == [NSNull null]) {
                     if (blocking < ANOMALY_ORANGE)
                         blocking = ANOMALY_ORANGE;
                 }
-                else if ([[json objectForKey:key] isEqualToString:@"blocked"]) {
+                else if ([[[json objectForKey:@"test_keys"] objectForKey:key] isEqualToString:@"blocked"]) {
```

Comment:
What happens if the value of the key is a boolean (which it can be)? Does calling `isEqualToString` cause some unhandled exception? (apologies for the noob obj-c question).

Reply:
according to the specifications I got it can't be, only string or null.
Or was your question more generic about what can happen in that case?

Reply:
Ah yes, you are right. I was confusing this with some other keys. In any case what happens if it isn't a string? If an exception is triggered, I think it would be wise to in any case be defensive and assert the type.

Reply:
The crash happens only in the NULL case.
If it's a bool it will be treated like "1" string